client/scripts/mifare_ad.lua

```diff
@@ -168,7 +167,7 @@ local function main( args)
 
 	local uid = open()
 	if (uid==nil) then oops("No card present") return end
-	print(("UID: %s"):format(uid))
+	print(("UID: %s"):format(uid.uid))
```

Comment:
I here I would suggest you move the code from "open()"  instead...    Since you basically are doing the same thing.